be/src/exec/olap_common.h

```diff
@@ -44,25 +45,26 @@ std::string cast_to_string(int8_t);
 /**
  * @brief Column's value range
  **/
-template <class T>
+template <doris::PrimitiveType primitive_type>
 class ColumnValueRange {
 public:
-    typedef typename std::set<T>::iterator iterator_type;
+    using CppType = typename doris::PrimitiveTypeTraits<primitive_type>::CppType;
+    using IteratorType = typename std::set<CppType>::iterator;
 
     ColumnValueRange();
 
-    ColumnValueRange(std::string col_name, PrimitiveType type);
+    ColumnValueRange(std::string col_name, doris::PrimitiveType type);
 
-    ColumnValueRange(std::string col_name, PrimitiveType type, const T& min, const T& max,
-                     bool contain_null);
+    ColumnValueRange(std::string col_name, doris::PrimitiveType type, const CppType& min,
```

Comment:
Maybe we can remove the arg `type` because it is the same as the template arg `doris::PrimitiveType primitive_type`?